app/views/queue_items/index.html.haml

```diff
@@ -4,29 +4,35 @@
       %article
         %header
           %h2 My Queue
-        %table.table
-          %thead
-            %tr
-              %th(width="10%") List Order
-              %th(width="30%") Video Title
-              %th(width="10%") Play
-              %th(width="20%") Rating
-              %th(width="15%") Genre
-              %th(width="15%") Remove
-          %tbody
-            - @items.each do |queue|
+        = form_tag(update_queue_path, method: :post) do
+          %table.table
+            %thead
               %tr
-                %td
-                  %input.form-control(type="text" value="#{queue.list_order}")
-                %td
-                  = link_to queue.video_title, video_path(queue.video)
-                %td
-                  = button_to "Play", nil, class: "btn btn-default"
-                %td= queue.rating
-                %td
-                  = link_to queue.category_name, queue.category
-                %td
-                  = link_to queue_item_path(queue), method: :delete do
-                    %i.glyphicon.glyphicon-remove
+                %th(width="10%") List Order
+                %th(width="30%") Video Title
+                %th(width="10%") Play
+                %th(width="20%") Rating
+                %th(width="15%") Genre
+                %th(width="15%") Remove
+            %tbody
+              - @items.each_with_index do |queue_item, index|
+                %tr
+                  %td
+                    = text_field_tag "queue_items_data[list_orders][]",
+                                     queue_item.list_order,
+                                     class: "form-control"
+                  %td
+                    = link_to queue_item.video_title, video_path(queue_item.video)
```

Comment:
I recommend using short-hand routes, `queue_item.video` instead of `video_path(queue_item.video)`